templates/_dialog_payment_info.tpl

```diff
@@ -32,7 +32,11 @@
         <tr>
             <th>{_ Name _}</th>
             <td>
-                {{ p.name_first|escape }} {{ p.name_surname_prefix|escape }} {{ p.name_surname|escape }}
+                {% if p.user_id %}
+                    <a href="{% url admin_edit_rsc id=p.user_id %}">{% include "_name.tpl" id=p.user_id %}</a>
+                {% else %}
+                    {{ p.name_first|escape }} {{ p.name_surname_prefix|escape }} {{ p.name_surname|escape }}
+                {% endif %}
```

Comment:
Not sure if it is wiser to always show what is in the payment table; we can see the user's name on their profile page anyway. This way any discrepancies are discoverable.